src/Handlebars.Net.Helpers/HandlebarsHelpers.cs

```diff
@@ -1,201 +1,203 @@
+        {
+            Guard.NotNull(optionsCallback, nameof(optionsCallback));
+
+            var options = new HandlebarsHelpersOptions();
+            optionsCallback(options);
+
+            if (options.CultureInfo != null)
+            {
+                handlebarsContext.Configuration.FormatProvider = options.CultureInfo;
+            }
+
+            // https://github.com/Handlebars-Net/Handlebars.Net#relaxedhelpernaming
+            handlebarsContext.Configuration.Compatibility.RelaxedHelperNaming = options.PrefixSeparatorIsDot;
+
+            var helpers = new Dictionary<Category, IHelpers>
+            {
+                { Category.Constants, new ConstantsHelpers(options) },
+                { Category.Enumerable, new EnumerableHelpers(options) },
+                { Category.Math, new MathHelpers(options) },
+                { Category.Regex, new RegexHelpers(options) },
+                { Category.String, new StringHelpers(options) },
+                { Category.Url, new UrlHelpers(options) }
+            };
+
+            foreach (var item in helpers.Where(h => options.Categories == null || options.Categories.Length == 0 || options.Categories.Contains(h.Key)))
+            {
+                var helper = item.Value;
+                Type helperClassType = helper.GetType();
+
+                var methods = helperClassType.GetMethods()
+                    .Select(methodInfo => (methodInfo, methodInfo.GetCustomAttribute<HandlebarsWriterAttribute>()))
+                    .Where(x => x.Item2 is { });
+
+                foreach (var x in methods)
+                {
+                    var name = GetName(x, options, item);
+
+                    RegisterHelper(options, handlebarsContext, helper, x.Item2.Type, x.methodInfo, name);
+                    RegisterBlockHelper(options, handlebarsContext, helper, x.methodInfo, name);
+                }
+            }
+        }
+
+        private static string GetName((MethodInfo methodInfo, HandlebarsWriterAttribute attribute) x, HandlebarsHelpersOptions options, KeyValuePair<Category, IHelpers> item)
+        {
+            var names = new List<string>();
+            if (x.attribute.Name is { } && !string.IsNullOrWhiteSpace(x.attribute.Name))
+            {
+                names.Add(x.attribute.Name);
+            }
+            else
+            {
+                if (options.Prefix is { } && !string.IsNullOrWhiteSpace(options.Prefix))
+                {
+                    names.Add(options.Prefix);
+                }
+
+                if (options.UseCategoryPrefix)
+                {
+                    names.Add(item.Key.ToString());
+                }
+
+                names.Add(x.methodInfo.Name);
+            }
+
+            return string.Join(options.PrefixSeparator, names);
+        }
+
+        private static void RegisterHelper(HandlebarsHelpersOptions helperOptions, IHandlebars handlebarsContext, object obj, WriterType writerType, MethodInfo methodInfo, string name)

```

Comment:
You should probably also register methods with return value.

Reply:
Can you explain this?

Reply:
My comment is related to this issue: https://github.com/Handlebars-Net/Handlebars.Net/issues/361
Some of the helpers should return value instead of writing it to output.

Reply:
thanks